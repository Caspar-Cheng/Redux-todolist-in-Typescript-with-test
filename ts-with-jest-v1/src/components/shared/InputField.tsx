import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';

interface InputProps {
  select: boolean;
  multiline: boolean;
  title: string;
  name: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const options = [
  { value: 'html', label: 'HTML' },
  { value: 'css', label: 'CSS' },
  { value: 'js', label: 'JS' },
];

const InputField: React.FC<InputProps> = ({
  select,
  multiline,
  title,
  name,
  value,
  onChange,
}: InputProps) => {
  return (
    <TextField
      select={select}
      multiline={multiline}
      label={title}
      name={name}
      margin="normal"
      value={value}
      onChange={onChange}
      style={{ width: 220 }}
    >
      {select &&
        options.map((option) => (
          <MenuItem key={option.value} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
    </TextField>
  );
};

export default InputField;
