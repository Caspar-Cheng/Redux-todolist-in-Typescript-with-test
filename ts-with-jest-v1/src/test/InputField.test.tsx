/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import renderer from 'react-test-renderer';
import InputField from '../components/shared/InputField';

const mockFunction = jest.fn((x) => x === 'value is changed');
const createTestProps = () => ({
  select: false,
  multiline: true,
  title: 'this is a test',
  name: 'test',
  value: 'value',
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => mockFunction(event),
});

describe('<InputField />', () => {
  describe('rendering', () => {
    const props = createTestProps();

    it('renders correctly', () => {
      const tree = renderer.create(<InputField {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should change the value to', () => {
      expect(mockFunction.mock.results.values).toBe('value is changed');
    });
  });
});
