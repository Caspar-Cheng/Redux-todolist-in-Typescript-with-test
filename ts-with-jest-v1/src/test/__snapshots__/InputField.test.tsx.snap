// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> rendering renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  style={
    Object {
      "width": 220,
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
  >
    this is a test
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
    onClick={[Function]}
  >
    <textarea
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
      disabled={false}
      name="test"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      rows={1}
      style={
        Object {
          "height": undefined,
          "overflow": null,
        }
      }
      value="value"
    />
    <textarea
      aria-hidden={true}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
      readOnly={true}
      style={
        Object {
          "height": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": "translateZ(0)",
          "visibility": "hidden",
        }
      }
      tabIndex={-1}
    />
  </div>
</div>
`;
